zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/FlowNodeInstanceFromIncidentHandler.java

```diff
@@ -59,29 +67,35 @@ public FlowNodeInstanceEntity createNewEntity(final String id) {
   public void updateEntity(
       final Record<IncidentRecordValue> record, final FlowNodeInstanceEntity entity) {
     final IncidentRecordValue recordValue = record.getValue();
-    entity
-        .setId(String.valueOf(recordValue.getElementInstanceKey()))
-        .setKey(recordValue.getElementInstanceKey())
-        .setPartitionId(record.getPartitionId())
-        .setFlowNodeId(recordValue.getElementId())
-        .setProcessInstanceKey(recordValue.getProcessInstanceKey())
-        .setProcessDefinitionKey(recordValue.getProcessDefinitionKey())
-        .setBpmnProcessId(recordValue.getBpmnProcessId())
-        .setTenantId(tenantOrDefault(recordValue.getTenantId()));
-
-    final String intentStr = record.getIntent().name();
-    if (intentStr.equals(IncidentIntent.CREATED.name())
-        || intentStr.equals(IncidentIntent.MIGRATED.name())) {
-      entity.setIncidentKey(record.getKey());
-    } else if (intentStr.equals(IncidentIntent.RESOLVED.name())) {
-      entity.setIncidentKey(null);
+    // build the treePath in the format
+    // <processInstanceKey>/<flowNodeInstanceKey>/.../<flowNodeInstanceKey>
+    // where upper level flowNodeInstanceKeys are normally subprocess(es) or multi-instance body
+    if (recordValue.getElementInstancePath() != null
+        && !recordValue.getElementInstancePath().isEmpty()) {
+      final Long currentProcessInstanceKey = Long.valueOf(entity.getId());
+      // call hierarchy for current process instance
+      final List<Long> elementInstancePath = recordValue.getElementInstancePath().getLast();
+      // we need to stop appending treePath on the current flowNodeInstanceKey
+      // e.g. incident returns call hierarchy [111,222,333] and we're currently building the
+      // treePath for
+      // flow node with id 222, then the treePath = 111/222
+      final int i = elementInstancePath.indexOf(currentProcessInstanceKey);
+      final List<String> treePathEntries =
+          elementInstancePath.stream().limit(i + 1).map(String::valueOf).toList();
+      entity.setTreePath(String.join("/", treePathEntries));
+      entity.setLevel(treePathEntries.size() - 1);
```

Comment:
So here we set only the intra path

Reply:
Where do we use the inter tree path?

Reply:
Inter tree path is used mainly when showing incident statistics on the diagram in Processes view. Also in post importer.